pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/dictionary/DictionaryIndexType.java

```diff
@@ -285,7 +302,15 @@ public static Dictionary read(PinotDataBuffer dataBuffer, ColumnMetadata metadat
     boolean loadOnHeap = indexConfig.isOnHeap();
     if (loadOnHeap) {
       String columnName = metadata.getColumnName();
-      LOGGER.info("Loading on-heap dictionary for column: {}", columnName);
+      OnHeapDictionaryConfig onHeapConfig = indexConfig.getOnHeapDictionaryConfig();
+      if (onHeapConfig != null && onHeapConfig.isEnableInterning()) {
+        _strInternerInfoMap.putIfAbsent(columnName, new FALFInterner<>(onHeapConfig.getInternerCapacity()));
```

Comment:
What if two different tables have a column with the same name and both have this feature enabled but different interner capacity?